### PR TITLE
blockPricesAlarm tests.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
This PR adds a test around the blockPricesAlarm and moves the active network gating of polling for block prices up to the alarm function rather than the `pollBlockPricesPerNetwork` itself.

### To Test
- [ ] Tests should pass
- [ ] Networks should correctly stop being queried for block prices after the `NETWORK_POLLING_TIMEOUT` has elapsed.

Latest build: [extension-builds-2348](https://github.com/tallyhowallet/extension/suites/10340092045/artifacts/509226135) (as of Thu, 12 Jan 2023 19:26:04 GMT).

### CODE REVIEW

File: background/services/chain/tests/index.unit.test.ts [hunk 1]
```diff
@@ -27,6 +29,30 @@ describe("Chain Service", () => {
...
+    describe("handleBlockPricesAlarm", () => {
+      it("should only poll for block prices on active networks", async () => {
+        const sixMinsAgo = Date.now() - 6 * MINUTE
+        ;(
+          chainService as unknown as ChainServiceExternalized
+        ).lastUserActivityOnNetwork = {
+          [ETHEREUM.chainID]: Date.now(),
+          [POLYGON.chainID]: sixMinsAgo,
+          [OPTIMISM.chainID]: sixMinsAgo,
+        }
```
Comment (greg-nagy) [inline background/services/chain/tests/index.unit.test.ts:42]:
```suggestion
        const ChainServiceExternalized =
          chainService as unknown as ChainServiceExternalized

        ChainServiceExternalized.lastUserActivityOnNetwork = {
          [ETHEREUM.chainID]: Date.now(),
          [POLYGON.chainID]: sixMinsAgo,
          [OPTIMISM.chainID]: sixMinsAgo,
        }
        ;[ETHEREUM, POLYGON, OPTIMISM].forEach((network: EVMNetwork) => {
          ChainServiceExternalized.subscribedNetworks.push({
            network,
            provider: {} as unknown as SerialFallbackProvider,
          })
        })
```

This breakds the indexing/tests/index.unit.test.ts` with `IndexingService › service start › should update cache once token lists load`. Probably there is some state persisting in the mocks. Didn't dig to the end.

**Review by greg-nagy — CHANGES_REQUESTED** [review]
I love these testing prs

This test as failing now, probably because of changes that came later in time.

**Comment by 0xDaedalus** [comment]
Have not forgotten about this - there are just a few more testing and alarm related PRs that I'd like to merge before diving in here

**Comment by greg-nagy** [comment]
The time is ripe 🍎 @0xDaedalus

**Comment by 0xDaedalus** [comment]
A bit too stale to fixup - will revisit in future

**Pull request closed without merging**